assets/javascripts/discourse/components/chat-composer-uploads.js

```diff
@@ -0,0 +1,172 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import UppyMediaOptimization from "discourse/lib/uppy-media-optimization-plugin";
+import ComposerUploadUppy from "discourse/mixins/composer-upload-uppy";
+import {
+  authorizedExtensions,
+  authorizesAllExtensions,
+} from "discourse/lib/uploads";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+
+export default Component.extend(ComposerUploadUppy, {
+  classNames: ["chat-composer-uploads"],
+
+  // represents the component state of uploads
+  _uploads: null,
+  inProgressUploads: null,
+  uploads: null,
+  uploadProcessorActions: null,
+  uploadPreProcessors: null,
+  uploadMarkdownResolvers: null,
+  uploadType: "chat-composer",
+  uppyId: "chat-composer-uppy",
+  fullPage: false,
+  composerModelContentKey: "value",
```

Comment:
Another possible candidate to drop for composer, it's only used in core to stop duplicate markdown for the upload placeholders:

https://github.com/discourse/discourse/blob/a0eddd0518ccd619a0a7661431cbaa7d6a93a98e/app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js#L452-L454

Reply:
Leaving this here but nevermind it, see the main PR comment.